src/main/scala/viper/carbon/modules/impls/DefaultExpModule.scala

```diff
@@ -391,8 +391,9 @@ class DefaultExpModule(val verifier: Verifier) extends ExpModule with Definednes
         def translate(e: sil.Exp): Seqn = {
           val checks = components map (_.partialCheckDefinedness(e, error, makeChecks = makeChecks))
           val stmt = checks map (_._1())
+
           // AS: note that some implementations of the definedness checks rely on the order of these calls (i.e. parent nodes are checked before children, and children *are* always checked after parents.
-          val stmt2 = for (sub <- e.subnodes if sub.isInstanceOf[sil.Exp]) yield {
+          val stmt2 = for (sub <- expSubnodesForDefinedness(e) if sub.isInstanceOf[sil.Exp]) yield {
```

Comment:
If you're filtering out everything that's not an expression anyway, I guess you could do that directly inside the new function and make it a ``subexpressionsForDefinedness`` function. Doesn't really change anything of course, but I think usually, the notion of well-definedness doesn't even really make sense for any subnodes that aren't expressions anyway.